src/arc_wake.rs

```diff
@@ -0,0 +1,78 @@
+// From htttps://github.com/rust-lang/rustc
```

Comment:
Since I pulled this file from rust-lang/rustc I included their LICENSE-MIT in the top of the file.

Reply:
Very good :)  However, the license also needs to be shipped with binary releases. We're not hosting any of those here, so technically this repository is fine. But if people use this crate as dependency, they'll almost certainly miss the license if it's not in the LICENSE file.

Could you add the license to the LICENSE file instead, below the BSD license, with a note that it applies to this file?

Reply:
The comment about the origin of the file is nice anyway, lets keep that in here.